accepted/2020/minimum-os-version/minimum-os-version.md

```diff
@@ -0,0 +1,245 @@
+
+## Assembly Attribute
+
+The minimum OS version will be burned into the assembly at build time using an assembly attribute. This attribute will be in the `System.Runtime.Versioning` namespace and will take the following form:
+
+```csharp
+[AttributeUsage(AttributeTargets.Assembly, AllowMultiple=true)]
+public sealed class OSMinimumVersion
+{
+    public OSMinimumVersion(
+    string platformIdentifier,
+    int majorVersion, int minorVersion, int servicingVersion
+    ) {}
+    public string PlatformIdentifier { get; }
+    public Version Version { get; }
+}
+```
+
+As with `PlatformAvailabilityAttribute` , the `PlatformIdentifier` property is included so that platform-agnostic assemblies can participate in platform minimum version annotations and checks.
+
+The MSBuild step that generates and injects TFM assembly attributes will be updated to generate this attribute when the `MinimumPlatformVersion` and `TargetPlatformIdentifier` properties are set.
+
+Platform-neutral assemblies, i.e. assemblies that target a platform-neutral TFM such as `net5`, will have to declare the attribute explicitly.
+
+> NOTE: This seems a bit inconsistent. As a user, I’d expect `Minimum{OSName}Version` properties to work in platform-neutral assemblies too, but handling all those properties when building platform-neutral assemblies would be problematic as it would centralize the definitions of all of the platform-specific `MinimumVersion` properties into the `Microsoft.Net.Sdk` targets.
+>
+>Perhaps we could use items and metadata instead of platform-specific properties, e.g. `<PlatformInfo Include="ios" MinimumVersion="15.0" />`. For platform-specific projects, we could allow the simpler `MinimumPlatformVersion` property by mapping it into one of these items, e.g. `<PlatformInfo Include="$(TargetPlatformIdentifier)" Version="$(MinimumPlatformVersion)" Condition="'$(TargetPlatformIdentifier)' != ''" />`.
+
+## NuGet Manifest
+
+When generating a NuGet package using MSBuild, any platform minimum version constraints expressed in the project file will be listed in the `.nuspec`  manifest in a `<platformInfo>` child of the `<metadata>` element. This element and all of its children are optional. They expose the same information as the assembly annotation attributes, but avoid the need for consumers to read the assembly metadata. This could potentially be validated and/or automated by NuGet..
+
+Within the `<platformInfo>` there are child elements corresponding to the TFM short names (used by NuGet as the folder name) of the reference assemblies in the NuGet. If the element represents a platform-specific TFM (e.g. `net5-ios14.0` or `net6-android9.0`) then it must have a `minimumVersion` attribute representing the minimum platform version for that platform:
+
+```xml
+<metadata>
+    <platformInfo>
+    <net5-ios15.0 minimumVersion="13.0" />
+    <net5-mac10.15 minimumVersion="10.11" />
+    </platformInfo>
+</metadata>
```

Comment:
```suggestion
<metadata>
    <platforms framework="net5.0">
      <platform id="ios" minVersion="13.0" targetVersion="13.2" maxVersion="14.0">
      <platform id="mac" minVersion="10.10" targetVersion="10.11" maxVersion="10.14">
      <platform id="android" minVersion="18" targetVersion="21" maxVersion="24">
    </platforms>
    <platforms framework="net6.0">
      <platform id="ios" minVersion="13.1" targetVersion="13.3" maxVersion="14.0">
      <platform id="mac" minVersion="10.11" targetVersion="10.12" maxVersion="10.14">
      <platform id="android" minVersion="19" targetVersion="22" maxVersion="24">
    </platforms>
</metadata>
```

We can use this format for the platform neutral declarations too (see **below**). These are only needed if we have interop between native lib and a managed implementation but could be useful for binding libs too as they are essentially more or less the same!

`targetVersion` and `maxVersion` are totally optional. They could used to further manage the dependencies and resolve conflicts for large typical project that targets multiple platforms.